checks/shell_download_validate.go

```diff
@@ -480,22 +482,28 @@ func isUnpinnedPipInstall(cmd []string) bool {
 			continue
 		}
 
-		// TODO(laurent): https://github.com/ossf/scorecard/pull/611#discussion_r660203476.
-		// Support -r <> --require-hashes.
+		// https://github.com/ossf/scorecard/issues/1306#issuecomment-974539197.
+		if strings.EqualFold(cmd[i], "--require-hashes") {
+			hashHashes = true
+			continue
+		}
 
 		// Exclude *.whl as they're mostly used
 		// for tests. See https://github.com/ossf/scorecard/pull/611.
 		if strings.HasSuffix(cmd[i], ".whl") {
 			hasWhl = true
+			// We continue because a command may contain
+			// multiple packages to install, not just `.whl` files.
 			continue
 		}
 
-		// Any other arguments are considered unpinned.
-		return true
+		// Other arguments are present.
+		hasAddArgs = true
 	}
 
-	// We get here only for `pip install [bla.whl ...]`.
-	return isInstall && !hasWhl
+	// We get here only for
+	// `pip install [bla.whl ...]`, `pip install <> --require-hashes`
+	return isInstall && !((hasWhl && !hasAddArgs) || hashHashes)
```

Comment:
This logic is getting a bit hard to reason about.

Maybe expand out some of these inversions into:

```
return isInstall &&  !hasHashes && (!hasWhl || hasAddArgs)
```

Or break these up into smaller conditions to make them easier to understand.

Reply:
I think this can just be:
```suggestion
	return isInstall && hasUnresolvedDeps
```